### ISSUE TITLE
aws-codepipeline-actions: SecretValue is not assignable to type

### ISSUE THREAD

**Issue by fire015** [aws/aws-cdk#12399]
Since 1.83.0 the `oauthToken` field in `codepipeline_actions.GitHubSourceAction` cannot be assigned the secret value as per the docs and examples.

### Reproduction Steps

```
new codepipeline_actions.GitHubSourceAction({
  actionName: "...",
  owner: "...",
  repo: "...",
  oauthToken: cdk.SecretValue.secretsManager("my-github-token"),
  output: sourceOutput,
  branch: "...",
});
```

### What did you expect to happen?

Oauth token is assigned correctly

### What actually happened?

```
⨯ Unable to compile TypeScript:
lib/pipeline-stack.ts:54:7 - error TS2322: Type 'import("/opt/code/aws/pipeline/node_modules/@aws-cdk/core/lib/secret-value").SecretValue' is not assignable to type 'import("/opt/code/aws/pipeline/node_modules/@aws-cdk/aws-cloudwatch/node_modules/@aws-cdk/core/lib/secret-value").SecretValue'.
  Types have separate declarations of a private property 'value'.

54       oauthToken: cdk.SecretValue.secretsManager('my-github-token'),
         ~~~~~~~~~~

  node_modules/@aws-cdk/aws-codepipeline-actions/lib/github/source-action.d.ts:117:14
    117     readonly oauthToken: SecretValue;
                     ~~~~~~~~~~
    The expected type comes from property 'oauthToken' which is declared here on type 'GitHubSourceActionProps'
```

### Environment

  - **CDK CLI Version  :** 1.83.0
  - **Framework Version:** 1.83.0
  - **Node.js Version:** 10.22.1
  - **OS               :** Ubuntu
  - **Language (Version):** TypeScript 

### Other

It looks like there are multiple definitions of `SecretValue` one within `aws-cloudwatch` and one within `cdk-core`.

--- 

This is :bug: Bug Report

**Comment by skinny85** [comment]
Hey @fire015 ,

you have two versions of the CDK libraries in your dependency closure. You can see it in the error message you posted:

```
lib/pipeline-stack.ts:54:7 - error TS2322: Type
'import("/opt/code/aws/pipeline/node_modules/@aws-cdk/core/lib/secret-value").SecretValue' is not assignable to type
'import("/opt/code/aws/pipeline/node_modules/@aws-cdk/aws-cloudwatch/node_modules/@aws-cdk/core/lib/secret-value").SecretValue'.
```

This is an (unfortunate) feature of TypeScript.

If you unify the versions of the libraries in your CDK package.json file, maybe remove `node_modules` directory, and run `npm/yarn install` again, it should work.

Thanks,
Adam

**Comment by fire015** [comment]
Hey Adam I did try this and get the same issue.

I decided to run this outside my machine on a later version of Node and similar issue...

https://repl.it/@fire015/AlarmedAthleticSites#stack.ts

If you run `npm run tsc` inside the shell you get same error.

**Comment by skinny85** [comment]
Yes, because you clearly have multiple version of the CDK libraries in your dependency closure.

From your package-lock.json file:

```
        "@aws-cdk/region-info": {
          "version": "1.82.0",
          "resolved": 

    "@aws-cdk/cloud-assembly-schema": {
      "version": "1.83.0",
```

**Comment by fire015** [comment]
Understood but I didn't add the lock file it's generated automatically, if you take package.json and run npm install locally you should also see the same issue.

**Comment by skinny85** [comment]
I don't think you'll get the same error if you remove your `package-lock.json` file. That's because you're using `^` to specify your dependencies, which means running `npm install` without a `package-lock.json` at different times will produce different results (based on what version of the CDK is currently the latest).

If you want to unblock yourself, I can see two easy ways:

1. Remove the `package-lock.json` file, and run `npm install` again.
2. Use the [`npm update` command](https://docs.npmjs.com/cli/v6/commands/npm-update) command to update the CDK packages you are depending on in your `package-lock.json` file.

Thanks,
Adam

**Comment by fire015** [comment]
I figured out what the problem was!

In my `package.json` file the `@aws-cdk/core` dependency was missing the ^ so it was fixed to 1.82 whereas the other packages were correctly installing 1.83.

Thanks for your help.
